core/src/main/scala/kafka/log/Log.scala

```diff
@@ -2232,6 +2232,8 @@ class Log(@volatile var dir: File,
    *   <li> Cleaner creates one or more new segments with suffix .cleaned and invokes replaceSegments().
    *        If broker crashes at this point, the clean-and-swap operation is aborted and
    *        the .cleaned files are deleted on recovery in loadSegments().
+   *   <li> If the new cleaned segments contain empty(zero-sized) ones, replaceSegments() tries to update high watermark and log start offset,
```

Comment:
The comment needs to be adjusted since we are no longer updating the high watermark.